web/components/roomConfig/GitlabRepoConfig.tsx

```diff
@@ -0,0 +1,167 @@
+        }
+    }, [ignoredHooks, setIgnoredHooks]);
+    const [newInstanceState, setNewInstanceState] = useState<GitLabRepoConnectionState|null>(null);
+
+
+
+    const canEdit = !existingConnection || (existingConnection?.canEdit ?? false);
+    const commandPrefixRef = createRef<HTMLInputElement>();
+
+    const onSaveCb = useCallback(() => {
+        onSave({
+            ...(existingConnection?.config || newInstanceState),
+            ignoreHooks: ignoredHooks as any[],
+            commandPrefix: commandPrefixRef.current.value,
+        });
+    }, [existingConnection, newInstanceState, ignoredHooks, commandPrefixRef, onSave]);
+    
+    return <div>
+        {!existingConnection && <ConnectionSearch api={api} onPicked={setNewInstanceState} />}
+        <InputField visible={!!existingConnection} label="GitLab Instance" noPadding={true}>
+            <input disabled={true} type="text" value={existingConnection?.config.instance} />
+        </InputField>
+        <InputField visible={!!existingConnection} label="Project" noPadding={true}>
+            <input disabled={true} type="text" value={existingConnection?.config.path} />
+        </InputField>
+        <InputField visible={!!existingConnection || !!newInstanceState} ref={commandPrefixRef} label="Command Prefix" noPadding={true}>
+            <input type="text" value={existingConnection?.config.commandPrefix || "!gl"} />
+        </InputField>
+        <InputField visible={!!existingConnection || !!newInstanceState} label="Events" noPadding={true}>
+            <p>Choose which event should send a notification to the room</p>
+            <ul>
+                <li><input type="checkbox" x-event-name="merge_request" checked={!ignoredHooks.includes("merge_request")} onChange={setIgnoredHook} />Merge requests</li>
+                <ul>
+                    <li><input disabled={ignoredHooks.includes("merge_request")} type="checkbox" x-event-name="merge_request.open" checked={!ignoredHooks.includes("merge_request.open")} onChange={setIgnoredHook} />Opened</li>
+                    <li><input disabled={ignoredHooks.includes("merge_request")} type="checkbox" x-event-name="merge_request.close" checked={!ignoredHooks.includes("merge_request.close")} onChange={setIgnoredHook} />Closed</li>
+                    <li><input disabled={ignoredHooks.includes("merge_request")} type="checkbox" x-event-name="merge_request.merge" checked={!ignoredHooks.includes("merge_request.merge")} onChange={setIgnoredHook} />Merged</li>
+                    <li><input disabled={ignoredHooks.includes("merge_request")} type="checkbox" x-event-name="merge_request.review" checked={!ignoredHooks.includes("merge_request.review")} onChange={setIgnoredHook} />Reviews</li>
+                </ul>
+                <li>
+                    <input type="checkbox" x-event-name="push" checked={!ignoredHooks.includes("push")} onChange={setIgnoredHook} />
+                    Pushes
```

Comment:
If you wrap these in `<label/>`s, clicking on the text will toggle the checkbox.

You may want to see if the wrapping label messes with your styles.

```suggestion
                    <label>
                        <input type="checkbox" x-event-name="push" checked={!ignoredHooks.includes("push")} onChange={setIgnoredHook} />
                        Pushes
                    </label>
```

Reply:
Done this, and also turned these into functional components to make this more readable.